data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -1319,14 +1319,513 @@ Marksman attacks are only affected if the chance to hit is greater than 60%."
+                            damage_type = impact
+                            kill = yes
+                            experience = yes
+                            fire_event = yes
+                            slowed = yes
+                            animate = yes
+                            [primary_attack]
+                                name = bolas
+                            [/primary_attack]
+                        [/harm_unit]
+                    [/then]
+                [/if]
+            [/do]
+        [/foreach]
+
+        {CLEAR_VARIABLE opportunist_hit_units,opportunist_hit_chance,opportunist_hit_roll}
+    [/event]
+#enddef
+
+#define WEAPON_SPECIAL_TAUNT
+    [dummy]
+        id = taunt
+        name = _"taunt"
+        description= _ "It taunts the enemy making it try to attack Kaleh no matter what."
+    [/dummy]
+#enddef
+
+#define WEAPON_SPECIAL_TAUNT_EVENTS
+    [event]
+        name = attack
+        first_time_only = no
+
+        [filter_attack]
+            special_id = taunt
+        [/filter_attack]
+
+        [object] 
+            [filter] 
+                find_in = second_unit
+            [/filter]
+
+            [effect]
+                apply_to = new_ability
+
+                [abilities]
+                    {ABILITY_TAUNTED}
+                [/abilities]
+            [/effect]
+        [/object]
+    [/event]
+#enddef
+
+#define ABILITY_TAUNTED
+    [dummy]
+        id = taunted
+        name = _"taunted"
+        description= _ "This unit is taunted by Kaleh, it will try to attack him no matter what."
+    [/dummy]
+#enddef
+
+#define ABILITY_INSPIRE
+    [leadership]
+        id=inspire
+        value="(25 * (level - other.level + 1))"
+        cumulative=no
+        name= _ "inspire"
+        female_name= _ "female^inspire"
+        description= _ "This unit can inspire own units that are next to it, making them fight better. Adjacent own units of lower or equal level will deal 25% more damage plus 25% for each level they are below the inspiring unit."
+        special_note=_"The presence of this unit inspires own units next to it to deal more damage in combat, though this only applies to units of lower or equal level."
+        affect_self=no
+        [affect_adjacent]
+            [filter]
+                formula="level <= other.level"
+            [/filter]
+        [/affect_adjacent]
+    [/leadership]
+#enddef
+
+#define STR_COOLDOWN_NAME_2
+_ "cooldown 1"#enddef
+
+#define STR_COOLDOWN_NAME_3
+_ "cooldown 2"#enddef
+
+#define STR_COOLDOWN_RESTRICT_2
+_ "After using this attack, you cant use it during your next turn"#enddef
+
+#define STR_COOLDOWN_RESTRICT_3
+_ "After using this attack, you cant use it during your next two turn"#enddef
+
+#define WEAPON_SPECIAL_COOLDOWN X
+    [disable]
+        id = cooldown_{X}
+        name = {STR_COOLDOWN_NAME_{X}}
+        description = {STR_COOLDOWN_RESTRICT_{X}}
+        [filter_self]
+            formula = "self.wml_vars.cooldown > 0"
```

Comment:
This copies the design of the song abilities - it's a per-weapon cooldown, but put in a per-unit variable. If two weapons have cooldowns, they'll conflict with each other.

Does this one have a lore reason why it can be used on defense even when it's on cooldown?

Reply:
So you have a suggestion on what to do about the weapon cooldown? Do you want to add [variables] in attacks? Or maybe use the name of the attack as a variable key?

Reply:
Documenting the limitation in comments would help a lot, even if it doesn't change the implementation.

Reply:
> Does this one have a lore reason why it can be used on defense even when it's on cooldown?

It cant, the wole attack using it cant be used on defense.